enterprise/frontend/src/embedding-sdk/components/public/InteractiveDashboard/InteractiveDashboard.tsx

```diff
@@ -131,20 +136,49 @@ const InteractiveDashboardInner = ({
 };
 
 export const InteractiveDashboard = renderOnlyInSdkProvider(
-  ({ dashboardId, ...rest }: InteractiveDashboardProps) => {
-    const { id, isLoading } = useValidatedEntityId({
+  ({ dashboardId: initialDashboardId, ...rest }: InteractiveDashboardProps) => {
+    const { id: resolvedDashboardId, isLoading } = useValidatedEntityId({
       type: "dashboard",
-      id: dashboardId,
+      id: initialDashboardId,
     });
 
+    const errorPage = useSdkSelector(getErrorPage);
+    const dispatch = useSdkDispatch();
+    useEffect(() => {
+      if (resolvedDashboardId) {
+        dispatch(setErrorPage(null));
+      }
+    }, [dispatch, resolvedDashboardId]);
```

Comment:
I wonder if there could be cases where other errors were showing first, and this `setErrorPage(null)` call would inadvertently clear that error?

Reply:
Good point. I don't think we've ever used this error yes anywhere in our SDK, so this should be safe. I was trying to get the info on the dashboard request, if there's a better to do that then I'll take it.

Reply:
Seems we're still using the old API (non-RTK query) for this request.

https://github.com/metabase/metabase/blob/7061c613220ce6a202a13f98d5d6990b42832c5a/frontend/src/metabase/services.js#L177

I was trying to use other dashboard state with no success